### ISSUE TITLE
Microk8s LXD installation with GPU enabled 

### ISSUE THREAD

**Issue by dmTriantaf** [canonical/microk8s#2926]
<------ LXD --- microk8s 1.23 --- nvidia GTX 1650 ti --- Ubuntu 20.04 ------>

I have an installation with lxd using profile as provided  https://raw.githubusercontent.com/ubuntu/microk8s/master/tests/lxc/microk8s.profile ,
leading to a privileged container. 
I' ve added gpu via lxc config and installed the same drivers with the host. nvidia-smi is working.

microk8s enable gpu  **is not working.** 

ERROR "failed to get sandbox runtime: no runtime for "nvidia" is configured

I also tried installing **gpu-operator** via helm with no luck
```
helm install --wait --generate-name \
     -n gpu-operator --create-namespace \
     nvidia/gpu-operator
```

**How can I get GPU working with microk8s??**
I' ve also added, docker, containerd and nvidia-docker2.

(I also faced the same issues installing microk8s at my host, without lxc, using containerd.)

After enabling GPU, I want to add kubeflow. Enabling kubeflow is not supported at microk8s 1.23.
Will I be able to install it with juju?

**Comment by ktsakalozos** [comment]
Hi @dmTriantaf, Kubeflow is available for Kubernetes up to v1.21. 

On the web I see a few articles on getting GPU passthrough on lxc containers but nothing on how to have the k8s nvidia operator working on lxc. Have you tried asking over at https://github.com/NVIDIA/gpu-operator/ ?

**Comment by dmTriantaf** [comment]
Hi @ktsakalozos . Thanks for the reply. 

I couldn't get "microk8s enable gpu" at my host working (without lxc) at version 1.21 either. Scrapping to some issues, it was suggested to upgrade to 1.22+ for enabling GPU.  

So, for a simple case where I install 1.21 directly on my host. Is something prerequisite before enabling GPU?

I will then proceed with the LXD installation.

**Comment by fzhan** [comment]
> Hi @dmTriantaf, Kubeflow is available for Kubernetes up to v1.21.
> 
> On the web I see a few articles on getting GPU passthrough on lxc containers but nothing on how to have the k8s nvidia operator working on lxc. Have you tried asking over at https://github.com/NVIDIA/gpu-operator/ ?

I've been trying to get gpu enabled on 1.22 (3006), still says "failed to get sandbox runtime: no runtime for "nvidia" is configured.
